### PR TITLE
Fix emoji offsets

### PR DESCRIPTION
**Pull request by rinoc**
Fixes regression in latest release for #3044 that causes cursor to get stuck in some emoji's.

#### What's the new behavior?
Arrowing through emoji's works as expected.

#### How does this change work?
I just added the code back from this PR #3044; it seems to solve the issue.

#### Have you checked that...?

- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)

### CODE REVIEW

File: packages/slate/src/interfaces/editor/queries/location.ts [hunk 1]
```diff
@@ -854,3 +853,146 @@ const getWordDistance = (text: string): number => {
...
+    code === 0x2642 || // male (♂)
+    code === 0x2640 || // female (♀)
+    code === 0x2620 || // scull (☠)
+    code === 0x2695 || // medical (⚕)
+    code === 0x2708 || // plane (✈️)
+    code === 0x25ef // large circle (◯)
+  )
+}
+
+const getCharOffset = (text: string, forward?: boolean): number => {
```
Comment (ianstormtaylor) [inline packages/slate/src/interfaces/editor/queries/location.ts]: This `forward` argument doesn't ever appear to be used?

Comment (rinoc) [reply packages/slate/src/interfaces/editor/queries/location.ts]: You're right! Removed it.

File: packages/slate/src/interfaces/editor/queries/location.ts [hunk 2]
```diff
@@ -854,3 +853,125 @@ const getWordDistance = (text: string): number => {
...
+
+const getCharOffset = (text: string): number => {
+  let offset = 0
+  // prev types:
+  // SURR: surrogate pair
+  // MOD: modifier (technically also surrogate pair)
+  // ZWJ: zero width joiner
+  // VAR: variation selector
+  // BMP: sequenceable character from basic multilingual plane
+  let prev = null
```
Comment (ianstormtaylor) [inline packages/slate/src/interfaces/editor/queries/location.ts]: Can you set the type here to get this to typecheck?

**Comment by ianstormtaylor** [comment]
Thank you @rinoc, I really appreciate you taking the time to port this one over. This is great.

**Pull request merged**